### PR TITLE
Issue 81

### PR DESCRIPTION
**Pull request by alrusdi**
8 more cards fixed. Please merge, but do not remove a branch

### CODE REVIEW

**Comment by vincentneko** [comment]
@bafolts can you please review as both alrusdi and I have worked on this merge ? Thanks

File: src/TestingUtils.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { Player } from "./Player";
```
Comment (bafolts) [inline src/TestingUtils.ts]: This should go in `tests/TestingUtils.ts` as the class is only needed for files in `tests/`.

File: src/cards/Birds.ts [hunk 2]
```diff
@@ -15,14 +15,30 @@ export class Birds implements IActionCard, IProjectCard {
...
+        return false;
+      }
       return game.getOxygenLevel() >= 13 - player.getRequirementsBonus(game);
     }
     public onGameEnd(player: Player) {
       player.victoryPoints += player.getResourcesOnCard(this);
     }
     public play(_player: Player, game: Game) {
       if (game.getPlayers().length == 1) return undefined;
+      if ( ! this.canPlay(_player, game)) {
```
Comment (bafolts) [inline src/cards/Birds.ts]: Calls to `canPlay` shouldn't be included in the call to `play`. It is assumed the card can be played once `play` is called.

File: src/cards/CloudSeeding.ts [hunk 3]
```diff
@@ -11,7 +11,18 @@ export class CloudSeeding implements IProjectCard {
...
+    private playersWithHeatProduction(game: Game): boolean {
+      for (const player of game.getPlayers()) {
+        if (player.heatProduction >= 1) return true;
+      }
+      return false;
+    }
+    
     public canPlay(player: Player, game: Game): boolean {
+
+      if ( ! this.playersWithHeatProduction(game)) return false;
```
Comment (bafolts) [inline src/cards/CloudSeeding.ts]: https://github.com/bafolts/terraforming-mars/blob/master/src/cards/CloudSeeding.ts#L30

I would return the array of players with heat production and only allow those players to be selected. With that change an error no longer needs to be thrown and `canPlay` can check if the array is empty.

**Pull request merged**